angrmanagement/ui/dialogs/load_plugins.py

```diff
@@ -98,19 +92,37 @@ def _init_widgets(self):
     #
 
     def _on_ok_clicked(self):
-        # TODO: Unload unchecked plugins. Load checked ones.
         list_items = self._installed_plugin_list.findItems('*', Qt.MatchWildcard)  # type: List[QPluginListWidgetItem]
         for i in list_items:
             checked = i.checkState() == Qt.Checked
 
-            if checked and i.plugin_class.__name__ not in self._pm.enabled_plugins.keys():
-                _l.info("Loading plugin: {}".format(i.plugin_class.get_display_name()))
-                plugin = self._pm.enable_plugin(i.plugin_class.__name__)
-            elif not checked and i.plugin_class.__name__ in self._pm.enabled_plugins.keys():
-                _l.info("Disabling plugin: {}".format(i.plugin_class.get_display_name()))
-                self._pm.disable_plugin(i.plugin_class.__name__)
+            if checked and self._pm.get_plugin_instance(i.plugin_class) is None:
+                self._pm.activate_plugin(i.plugin_class)
+            elif not checked and self._pm.get_plugin_instance(i.plugin_class) is not None:
+                self._pm.deactivate_plugin(i.plugin_class)
+            else:
+                self._pm.load_plugin(i.plugin_class)
 
         self.close()
 
     def _on_cancel_clicked(self):
         self.close()
+
+    def _on_load_clicked(self):
+        file_path, _ = QFileDialog.getOpenFileName(
+            self, "Open a plugin (select __init__.py for packages)", "", "Python files (*.py)")
+        plugins = load_plugins_from_file(file_path)
+
+        if not plugins:
```

Comment:
This error is also shown after hitting "Cancel" on the Load Plugins dialog.

Reply:
Huh. The Qt documentation implies that hitting cancel will return None but it instead returns an empty string. I added a check for falsey results.